janusgraph-core/src/main/java/org/janusgraph/diskstorage/configuration/builder/ReadConfigurationBuilder.java

```diff
@@ -67,12 +67,12 @@ public ReadConfiguration buildGlobalConfiguration(ReadConfiguration localConfig,
             ModifiableConfiguration globalWrite = modifiableConfigurationBuilder.buildGlobalWrite(keyColumnValueStoreConfiguration);
 
             if (!globalWrite.isFrozen()) {
+                setupTimestampProvider(globalWrite, localBasicConfiguration, storeManager);
```

Comment:
Can you help me understand why this line must be put before `globalWrite.setAll(getGlobalSubset(localBasicConfiguration.getAll()));`?

By looking at the code, it seems to me that the order of these two lines does not matter.

Reply:
Yesterday I thought I was seeing the difference. For some reason I thought that the value from `localBasicConfiguration` was overwritten by `TIMESTAMP_PROVIDER.getDefaultValue()` but now checking again I'm trying to understand the difference and I don't see it anymore. Probably I missed something.
@BrunoBerisso could you please explain the difference in the behavior when we moved this line before `globalWrite.setAll(getGlobalSubset(localBasicConfiguration.getAll()));`?

Reply:
Sorry, this change doesn't fix #2590 I didn't test them properly before opening the PR.

I'll update the fix to use the correct`TIMESTAMP_PROVIDER` when creating the `KCVSConfiguration` instance used to write these settings.

Reply:
Sure, I've dismissed approvals in case of an accidental merge.